common/http/http.cpp

```diff
@@ -112,6 +112,12 @@ error::Error BreakDownUrl(const string &url, BrokenDownUrl &address) {
 		address.path = tmp.substr(split_index);
 	}
 
+	if (address.host.find("@") != string::npos) {
+		return error::Error(
+			make_error_condition(errc::not_supported),
+			"URL Username and password is not supported");
+	}
```

Comment:
I know it shouldn't really happen, but this makes me think about `address` being half- or even all-set when this error is returned. And thus can then easily be used for requests. While I have some doubts that all the code that calls this (or `SetAddress()`) actually checks the returned value (error) and clears the fields in case of error.

Reply:
Good point. I'll add this to all the error returns.